Mass extinction of open issues. (Lots of bug and QoL fixes)

Claycorp opened the pull request:
- Fixed #436
- Fixed #554
- Fixed #707 
- Fixed #759 
- Fixed #775 (sorta, This is a deviation for the original idea of a scale down but people liked it.)
- Fixed #799 
- Fixed #808 
- Fixed #933 
- Fixed #918 
- Fixed #939 
- Fixed #946 (Other mods may cause this issue too, it was fucking annoying though. Should help Could be fixed with CT scripts in other places more than likely.)
- Fixed #969 (69 lamo)
- Fixed #971 
- Fixed #988 
- Fixed #1072 
- Fixed #1036 
- Fixed #1068 
- Fixed #1038 
- Fixed #1050 
- Fixed #1054 
- Fixed #1067 
- Fixed #1097 
- Fixed #1095 

Misc non-issue changes.
- Javelins are infinite for creative users.
- Improved thrown javelin rotation.
- Added XZ deviations to all saplings.
- Metal Buckets don't empty in creative anymore.
- Barrels will not hold the contents they had anymore when sealed unless it's a valid recipe. (You can't hide the cheese from me...)
- Campfires aren't wusses anymore and won't disappear from a simple slap.

Claycorp committed:
Re-add the option to disable ingot piles. Closes #436

Claycorp committed:
Fixed javelins rendering incorrectly when thrown to the north or south, Also generally improved thrown rotations and positions. Closes #554
Added infinite javelins for creative users.

Claycorp committed:
Added feather plucking and matching damage source with configs. Closes #707

Claycorp committed:
Fixed farmland differences from vanilla and fixed fast lighting. Closes #759

Claycorp committed:
Changed TFC crops and saplings to use an offset like grass and flowers. Sorta Fixes #775

Claycorp committed:
Missed Dead crops in last bit.

Claycorp committed:
Added config option to disable all death of crops. Closes #799

Claycorp committed:
Added thirst potion effect. Closes #808

Claycorp committed:
Changed getRenderLayer to CUTOUT so it renders correctly. Closes #933

Claycorp committed:
Removed log drops from a good fisting and added config. Closes #918

Claycorp committed:
Added Knives as tools for breaking vine and leaf materials. Closes #939

Claycorp committed:
Fixed things keeping empty NBT tags when they shouldn't be. Closes #946
Made Metal buckets act correctly in creative when pouring.

Claycorp committed:
Added Sapling drop config. Closes #969 (heh 69 lmao)

Claycorp committed:
Removed extra plant sustainability options for vanilla based crops left from debug days. Closes #971

Claycorp committed:
Changed lit log pile to lit forge texture. Closes #988
Changed the fucking water name for stupid "hot" water to reflect what it really is... Closes #1072 (No we don't need more dammed waters..)
Fixed an oops with the change for logs from a fisting trees. Really Closes #918

Claycorp committed:
Barrels will not hold contents anymore if there is no valid recipe. #BlameDries Closes nothing cause it never made it to an issue. #StoppingTheCheese

Claycorp committed:
Improved some messages on strip world command. Closes #1036

Claycorp committed:
Added unfired recycle recipes. Closes #1068

Claycorp committed:
Fixed red apple sapling using banana texture. Closes #1038

Claycorp committed:
Added new ore value tooltip system including stack total values. Closes #1050

Claycorp committed:
Chonkafied fire pit hotboxes when things are attached to them so they are easier to interact with. Closes #1054
Made fire pits slightly harder so they don't get yeeted by a slight graze.

Claycorp committed:
Added back blue bed recipe. Closes #1067

Claycorp committed:
FIXED THE DAM FENCE GATES! Closes #1097

Claycorp committed:
Removed the need to drop everything to collect a placed item, Because we all know that one hand works just as well. Closes #1095

DisasterMoo commented on src/main/java/net/dries007/tfc/command/CommandStripWorld.java:
```diff
@@ -40,13 +40,13 @@ public String getName()
     {
-        return "/stripworld <radius> -> [DANGER] Strips all stone variant blocks + water.";
+        return "/stripworld <radiusInBlocks> -> [DANGER] Strips all stone variant blocks + water.";
     }
 
     @Override
     public void execute(MinecraftServer server, ICommandSender sender, String[] args) throws CommandException
     {
-        if (args.length != 1) throw new WrongUsageException("1 argument required.");
+        if (args.length != 1) throw new WrongUsageException("/stripworld <radiusInBlocks>");
```
Please use lang entries, refer to `CommandFindVeins` and it's lang entries if needed.

Claycorp replied:
I just edited what was there, I can update that to match though.

DisasterMoo commented on src/main/java/net/dries007/tfc/objects/potioneffects/PotionThirst.java:
```diff
@@ -0,0 +1,50 @@
+    @Override
+    public boolean isReady(int duration, int amplifier)
+    {
+        return true;
```
Minor nitpick: I think it would be better to do like hunger and drains the thirst over time instead of instantly.

Claycorp replied:
This is exactly like hunger? It's a copy paste of the hunger debuff set with a higher default (one order higher) so you don't need a 40 modifier on it to notice the drop.

Edit: Helps if I read the thing you highlight.
Testing it gives the same effect as hunger. I don't know why this option even exists as it does nothing. The only thing that uses it is health.

DisasterMoo commented on src/main/java/net/dries007/tfc/CommonEventHandler.java:
```diff
@@ -807,6 +810,30 @@ public static void pickupQuiverItems(EntityItemPickupEvent event) //only pickups
+                }
+                if (target instanceof EntityAnimalTFC)
+                {
+                    ((EntityAnimalTFC) target).setFamiliarity(((EntityAnimalTFC) target).getFamiliarity() - 0.4f);
```
If this is the intention you can mark this as resolved, but 0.4f = 40% of the total familiarity. Shouldn't it be 4% instead?

Claycorp replied:
Correct. Bad conversion from classic math to %. Should be .04!

DisasterMoo commented on src/main/java/net/dries007/tfc/ConfigTFC.java:
```diff
@@ -475,6 +501,16 @@ public static void onConfigChangedEvent(ConfigChangedEvent.OnConfigChangedEvent
+            "2 = Stack unit total info.",
+            "3 = All info."
+        })
+        @Config.LangKey("config." + MOD_ID + ".general.oreTooltipMode")
```
Missing lang entry

DisasterMoo commented on src/main/java/net/dries007/tfc/ConfigTFC.java:
```diff
@@ -426,6 +426,32 @@ public static void onConfigChangedEvent(ConfigChangedEvent.OnConfigChangedEvent
         @Config.Comment("Which inventory slots will ammo refill/pickup search for quivers? none = nowhere, they're just extra inventory spaces; armor = only in armor slots; hotbar = add hotbar; main = add main inventory.")
         @Config.LangKey("config." + MOD_ID + ".general.quiverSearch")
         public String quiverSearch = "hotbar";
+
+        @Config.Comment("Disable the ability to create ingot piles?")
+        @Config.LangKey("config." + MOD_ID + ".general.disableIngotPile")
+        public boolean placeIngotPiles = false;
+
+        @Config.Comment("Entities that can be plucked for feathers.")
+        @Config.LangKey("config." + MOD_ID + ".general.pluckableEntities")
+        public String[] pluckableEntities = new String[] {"tfc:chickentfc", "tfc:pheasanttfc", "tfc:parrottfc", "tfc:ducktfc"};
+
+        @Config.Comment("Damage dealt to an entity when a feather is harvested.")
+        @Config.RangeDouble(min = 0)
+        @Config.LangKey("config." + MOD_ID + ".general.damagePerFeather")
+        public double damagePerFeather = 0.6;
+
+        @Config.Comment("If true, crops will never die under any circumstances. THIS DOES NOT MEAN THEY WILL ALWAYS GROW!")
+        @Config.LangKey("config." + MOD_ID + ".general.shouldCropsDie")
+        public boolean shouldCropsDie = true;
+
+        @Config.Comment("If false, logs will drop logs when punched.")
+        @Config.LangKey("config." + MOD_ID + ".general.doLogsRequireAxe")
+        public boolean doLogsRequireAxe = true;
+
+        @Config.Comment("If false, leaves will not drop saplings.")
+        @Config.LangKey("config." + MOD_ID + ".general.doLeavesDropSaplings")
+        public boolean doLeavesDropSaplings = true;
+
```
Missing lang entries

DisasterMoo reviewed (changes requested):
Need a couple changes, most notably the missing lang entries, but otherwise seems good.

alcatrazEscapee commented on src/main/java/net/dries007/tfc/objects/blocks/devices/BlockFirePit.java:
```diff
@@ -56,6 +56,7 @@
 
     private static final AxisAlignedBB DEFAULT_AABB = new AxisAlignedBB(0, 0, 0, 1, 0.125, 1);
     private static final AxisAlignedBB FIREPIT_AABB = new AxisAlignedBB(0, 0, 0, 1, 0.125, 1);
+    private static final AxisAlignedBB FIREPIT_WITH_SHIT_AABB = new AxisAlignedBB(0, 0, 0, 1, 0.7, 1);
```
```suggestion
    private static final AxisAlignedBB FIREPIT_WITH_ATTACHMENT_AABB = new AxisAlignedBB(0, 0, 0, 1, 0.7, 1);
```
Can we not?

alcatrazEscapee commented on src/main/java/net/dries007/tfc/objects/items/metal/ItemMetalBucket.java:
```diff
@@ -124,6 +124,10 @@ else if (fluid.getBlock() instanceof BlockLiquid)
                             }
                         }
                         worldIn.playSound(null, pos, SoundEvents.ITEM_BUCKET_EMPTY, SoundCategory.PLAYERS, 1.0F, 1.0F);
+                        if (stack.hasTagCompound() && stack.getTagCompound().isEmpty())
+                        {
+                            stack.setTagCompound(null);
+                        }
```
So I understand the issue that this is trying to solve, but this is a really hack and slash way of getting around it and doesn't resolve the root issue. There are still plenty of situations where the NBT tag would still apply. As I mentioned in the issue, we need to tackle the root problem, namely, the situations in `FluidHandlerItemStackSimple` which saves the fluid to NBT, is not the same assumption that we operate under (that the fluid is never saved to NBT). For consistency with the rest of the mod we should go with the latter, but for simplicity it might be better to go with the former.

alcatrazEscapee commented on src/main/java/net/dries007/tfc/objects/items/wood/ItemWoodenBucket.java:
```diff
@@ -146,6 +146,10 @@ else if (fluid.getBlock() instanceof BlockLiquid)
                             }
                         }
                         worldIn.playSound(null, pos, SoundEvents.ITEM_BUCKET_EMPTY, SoundCategory.PLAYERS, 1.0F, 1.0F);
+                        if (stack.hasTagCompound() && stack.getTagCompound().isEmpty())
+                        {
+                            stack.setTagCompound(null);
+                        }
```
See above.

alcatrazEscapee commented on src/main/java/net/dries007/tfc/objects/recipes/ShapelessFluidFoodRecipe.java:
```diff
@@ -59,7 +59,12 @@ public ShapelessFluidFoodRecipe(ResourceLocation group, NonNullList<Ingredient>
             {
                 handler.drain(Fluid.BUCKET_VOLUME, true);
-                ret.set(i, handler.getContainer().copy());
+                ItemStack updatedItem = handler.getContainer().copy();
+                if (updatedItem.hasTagCompound() && updatedItem.getTagCompound().isEmpty())
+                {
+                    updatedItem.setTagCompound(null);
+                }
+                ret.set(i, updatedItem);
```
See above.

Claycorp committed:
Fixed bad math on feather plucking for reducing familiarity was 40% should be 4%.
Cleaned up AABB boxes on firepit.
"Fix this to use translations like other commands" they say, *commands have 3 separate layouts and a mess of translations* Y dis like dis...? The Italians called they want there spaghetti back, I have mailed it back to them via the internet.
Looked at the "hacky fix that should be proper". no idea, Blame forge. This works for now and people will stop complaining.

Claycorp committed:
Fixed Potion thing because apparently command doesn't do stuff items do and moo did the other thing wrong so I fixed that.
